common.go

```diff
@@ -0,0 +1,37 @@
+package set
+
+// CommonSet is the interface that all sets implement
+type CommonSet[T any] interface {
+	Slice() []T
+	Insert(T) bool
+	InsertSlice([]T) bool
+	Size() int
+	// ForEach  will call the callback function for each element in the set
+	// If the callback returns false, the iteration will stop
+	ForEach(call func(T) bool)
+}
+
+// InsertSliceFunc inserts all elements from the slice into the set
+func InsertSliceFunc[T, E any](set CommonSet[T], items []E, f func(element E) T) {
+	for _, item := range items {
+		set.Insert(f(item))
+	}
+}
+
+// Transform transforms the set a into another set b
+func Transform[T, E any](a CommonSet[T], b CommonSet[E], transform func(T) E) {
+	a.ForEach(func(item T) bool {
+		_ = b.Insert(transform(item))
+		return true
+	})
+}
```

Comment:
Would it be possible to have two versions of this function? One that transforms into an existing set like this implementation, but also one that creates the new set? e.g.

```go
TransformUnion[T, E any](a CommonSet[T], b CommonSet[E], f func(T) E)
```

```go
Transform[T, E any](s CommonSet[T], f func(T) E) CommonSet[E]
```

Reply:
I'm to rename transform as TransformUnion.
The other transform is not implemented for now, it may have to be implemented next time.